### ISSUE TITLE
[Feature request]:  gordo exposes fixing the seed to the user

### ISSUE THREAD

**Issue by ruimaximo** [equinor/gordo#452]
At the moment gordo does not support fixing the seed in order to reproduce results.

Example:
This code does not fix the seed. 

`numpy.random.seed(42)`
`tensorflow.set_random_seed(42)`

If you re train a gordo model, it will produce different results.

The ability to reproduce results is important to study the influence of the hyperparameters on a model.

**Comment by flikka** [comment]
Discussion 15/10-2019:
It should be possible to get reproducible results today, but the two lines above is not enough.

Solution: We add a possibility to provide a seed, thus making subsequent model buildings provide the same model. This way we can both do hyperparameter tuning without worrying for the effects of the seed, as well as doing hyperparameter experiments on the seed to study the "stability".

If you don't provide the seed -> what should the behaviour be? Should we then fix it under the hood, or keep it as it is today?

**Comment by ruimaximo** [comment]
If no seed is provided then my suggestion is to use the random by default.
